internal/commands/app/create_inputs_test.go

```diff
@@ -709,6 +709,92 @@ func TestAppCreateInputsResolveCluster(t *testing.T) {
 		assert.Equal(t, errors.New("client error"), err)
 		assert.Equal(t, "123", expectedGroupID)
 	})
+
+	t.Run("should error when more than one cluster passed in when creating template app", func(t *testing.T) {
+		_, ui := mock.NewUI()
+		clusterNames := []string{"Cluster0", "Cluster1"}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{
+				{ID: "456", Name: clusterNames[0]},
+			}, nil
+		}
+		inputs := createInputs{
+			newAppInputs: newAppInputs{
+				Template: "ios.template.todo",
+			},
+			Clusters: clusterNames,
+		}
+		_, _, err := inputs.resolveClusters(ui, ac, "123")
+		assert.Equal(t, "template apps can only be created with one data source", err.Error())
+	})
+
+	t.Run("should error if no atlas clusters exist when creating template app", func(t *testing.T) {
+		_, ui := mock.NewUI()
+		clusterNames := []string{"Cluster0"}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{}, nil
+		}
+		inputs := createInputs{
+			newAppInputs: newAppInputs{
+				Template: "ios.template.todo",
+			},
+			Clusters: clusterNames,
+		}
+		_, _, err := inputs.resolveClusters(ui, ac, "123")
+		assert.Equal(t, "please create an atlas cluster before creating a template app", err.Error())
+	})
+
+	t.Run("should resolve a single data source if creating template app", func(t *testing.T) {
+		_, ui := mock.NewUI()
+		clusterNames := []string{"Cluster0"}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{
+				{ID: "1", Name: clusterNames[0]},
+				{ID: "2", Name: "Cluster1"},
+			}, nil
+		}
+		inputs := createInputs{
+			newAppInputs: newAppInputs{
+				Template: "ios.template.todo",
+			},
+			Clusters: clusterNames,
+		}
+
+		clusters, _, err := inputs.resolveClusters(ui, ac, "123")
+		assert.Equal(t, nil, err)
+		assert.Equal(t, 1, len(clusters))
+		assert.Equal(t, "Cluster0", clusters[0].Config.ClusterName)
+		assert.Equal(t, "mongodb-atlas", clusters[0].Name)
+	})
+
+	t.Run("should force mongodb-atlas as template data source name", func(t *testing.T) {
+		_, ui := mock.NewUI()
+		clusterNames := []string{"Cluster0"}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{
+				{ID: "456", Name: clusterNames[0]},
+			}, nil
+		}
+		inputs := createInputs{
+			newAppInputs: newAppInputs{
+				Template: "ios.template.todo",
+			},
+			Clusters:            clusterNames,
+			ClusterServiceNames: []string{"overridden_name"},
```

Comment:
just thinking about this now, it would be nice to print a warning message about `"overridden_name"` being ignored when this happens